__tests__/strategies/learning-steps.test.ts

```diff
@@ -0,0 +1,289 @@
+      [Rating.Hard]: { scheduled_minutes: 15, next_step: 1 },
+    })
+
+    result = BasicLearningStepsStrategy(params, State.Relearning, 2)
+    expect(result).toEqual({})
+  })
+})
+
+describe('integrated to FSRS', () => {
+  it('First learning step', () => {
+    const f = fsrs({
+      learning_steps: ['1m', '10m'],
+      relearning_steps: ['10m'],
+    })
+    const now = new Date(2022, 11, 29, 12, 30, 0, 0)
+    const card = createEmptyCard(now)
+    const record = f.repeat(card, now)
+    expect(record[Rating.Again].card.learning_steps).toEqual(0)
+    expect(record[Rating.Again].card.due.getTime() - now.getTime()).toEqual(
+      1000 * 60 * 1 // 1m
+    )
+
+    expect(record[Rating.Hard].card.learning_steps).toEqual(0)
+    expect(record[Rating.Hard].card.due.getTime() - now.getTime()).toEqual(
+      1000 * 60 * 6 // 6m=(1+10)/2
+    )
+
+    expect(record[Rating.Good].card.learning_steps).toEqual(1)
+    expect(record[Rating.Good].card.due.getTime() - now.getTime()).toEqual(
+      1000 * 60 * 10 // 10m
+    )
+  })
+
+  describe('Hardcoded learning steps', () => {
+    const learningStepsStrategy: TLearningStepsStrategy = (
+      params: FSRSParameters,
+      state: State,
+      cur_step: number
+    ) => {
+      if (cur_step >= 1) {
+        return {
+          [Rating.Again]: { scheduled_minutes: 5, next_step: 1 },
+          [Rating.Hard]: { scheduled_minutes: 10, next_step: 1 },
+        }
+      } else {
+        return {
+          [Rating.Again]: { scheduled_minutes: 1, next_step: 1 },
+          [Rating.Hard]: { scheduled_minutes: 5, next_step: 1 },
+          [Rating.Good]: { scheduled_minutes: 10, next_step: 1 },
+        }
+      }
+    }
+    it('use LearningStepsStrategy', () => {
+      const f = fsrs().useStrategy(
+        StrategyMode.LEARNING_STEPS,
+        learningStepsStrategy
+      )
+
+      expect(f['strategyHandler'].get(StrategyMode.LEARNING_STEPS)).toBe(
+        learningStepsStrategy
+      )
+
+      f.clearStrategy()
+      expect(
+        f['strategyHandler'].get(StrategyMode.LEARNING_STEPS)
+      ).toBeUndefined()
+    })
+
+    it('learning step', () => {
+      const f = fsrs({}).useStrategy(
+        StrategyMode.LEARNING_STEPS,
+        learningStepsStrategy
+      )
+      let now = new Date(2022, 11, 29, 12, 30, 0, 0)
+      let card = createEmptyCard(now)
+      let record = f.repeat(card, now)
+      // new
+      expect(record[Rating.Again].card.learning_steps).toEqual(1)
+      expect(record[Rating.Again].card.due.getTime() - now.getTime()).toEqual(
+        1000 * 60 * 1 // 1m
+      )
+
+      expect(record[Rating.Hard].card.learning_steps).toEqual(1)
+      expect(record[Rating.Hard].card.due.getTime() - now.getTime()).toEqual(
+        1000 * 60 * 5 // 5m
+      )
+
+      expect(record[Rating.Good].card.learning_steps).toEqual(1)
+      expect(record[Rating.Good].card.due.getTime() - now.getTime()).toEqual(
+        1000 * 60 * 10 // 10m
+      )
+      // learning
```

Comment:
Check that easy skips learning steps here if you want it.
```suggestion
      expect(record[Rating.Easy].card.learning_steps).toEqual(0)
      expect(dateDiffInDays(now, record[Rating.Easy].card.due)).toBe(Math.floor(default_w[3]))
      // learning
```